### PR TITLE
Nvm to json

### LINKED ISSUE #5: convert 'rgb' from .nvm to .json
be able to convert the rgb field from the .vnm Visual SfM format to the openSfM .json format

### PR DESCRIPTION
**Pull request by awkbr549**
Fixes #5 , #7 , and #8 . Prompts for input file and output file, and since this only currently does NVM to JSON, an extra file is needed. This can be attained by running OpenSfM's extract_metadata program (manual modification of OpenSfM bin directory required, ask for more details).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,50 @@
 # nvmconverter
 Converts the VisualSFM .nvm (N-View Match) filetype to the OpenSFM .json (JavaScript Object Notation) file type. The NVM file type is discribed here: [vsfm nvm](http://ccwu.me/vsfm/doc.html#nvm)
 
+# OpenSfM
+There is a Docker image of OpenSfM available by searching 
+`docker search awkbr549/opensfm'
 
+To run the image, it is advised to run with the following options:
+`-it'		--> used with the `run' command, creates an interactive and pseudo-tty connection to the container
```
Comment (nonbinaryprogrammer) [inline README.md:9]: fix markdown formatting here

File: README.md [hunk 2]
```diff
@@ -1,4 +1,50 @@
...
+`-ai'		--> used with the `start' command to allow interaction with an already existing Docker container
+`-p 8000:8000'		--> maps Docker container port 8000 to localhost 8000
+`-v dir/on/local/computer:dir/on/container' 	--> maps local files to the container
```
Comment (nonbinaryprogrammer) [inline README.md:12]: and the above three lines

File: README.md [hunk 3]
```diff
@@ -1,4 +1,50 @@
...
+
+On Windows, Docker's option to map local files to the container doesn't work (at least for me). In order to get around this, you can just use the `scp' file transfer command to move files from your local machine to a server, then also in your Docker container use `scp' to transfer files from the server to the container. Though cumbersome, this is the only workaround I've found.
+
+Example command:
+`docker run -it -p 8000:8000 awkbr549/opensfm:v2' //for new containers (!!! should only run once)
```
Comment (nonbinaryprogrammer) [inline README.md:17]: ok, it looks like you just consistently have the error of starting a code block with ` and ending it with '

File: convert.py [hunk 4]
```diff
@@ -9,40 +9,42 @@
...
-	input()
-	sys.exit(0)
+from arguments import *
+argArray = parseArguments()
 
+inputFile = ""
+outputFile = ""
+while True:
+	inputFile = raw_input("What file would you like to read from?\n--> ")
+	if (not os.path.isfile(inputFile)):
```
Comment (nonbinaryprogrammer) [inline convert.py]: check that the input file is of a valid type as well

File: convert.py [hunk 5]
```diff
@@ -9,40 +9,42 @@
...
+		print "ERROR: File does not exist."
+	else: break
+while True:
+	outputFile = raw_input("What file would you like to write to?\n--> ")
+	if (not (outputFile.endswith('.nvm') or outputFile.endswith('.json')) ):
```
Comment (nonbinaryprogrammer) [inline convert.py]: what if the file exists? will you overwrite it?

might want to prompt for if the file exists, "do you want to overwrite" type thing

File: convert.py [hunk 6]
```diff
@@ -9,40 +9,42 @@
...
+		print "ERROR: File type not supported."
+	else: break
 #Converting from NVM to something else
 if inputFile.endswith(".nvm"):
```
Comment (nonbinaryprogrammer) [inline convert.py:40]: add 'else' blocks to this and the if block in line 39

Comment (awkbr549) [inline convert.py:40]: I didn't add 'else' blocks because it doesn't do anything if there isn't the verbose option.

**Pull request merged**